core/src/main/java/com/aem/demo/core/models/MainPageModel.java

```diff
@@ -50,83 +48,50 @@ public class MainPageModel {
         while (pageIterator.hasNext()) {
             Page currentPage = pageIterator.next();
             Resource jcrContent = currentPage.getContentResource();
-            holdModels.add(jcrContent.adaptTo(NewsCardModel.class));
-        }
-
-        String searchText = Optional.ofNullable(servletRequest).map(SlingHttpServletRequest::getRequestParameterMap)
-                .map(map -> map.getValue("search"))
-                .map(RequestParameter::getString)
-                .orElse(null);
-
-
-        String paginationNumber = Optional.ofNullable(servletRequest).map(SlingHttpServletRequest::getRequestParameterMap)
-                .map(map -> map.getValue("page"))
-                .map(RequestParameter::getString)
-                .orElse(null);
-
-        isNumeric(paginationNumber);
-
-
-        if (searchText != null) {
-            models = searchService.retrieveModels(searchText);
-            paginateModels(models);
-            if (paginationNumber != null) {
-                models = separateModels(models, Integer.parseInt(paginationNumber));
-            }
-        } else if (paginationNumber != null) {
-            paginateModels(holdModels);
-            models = separateModels(holdModels, Integer.parseInt(paginationNumber));
-        } else {
-            paginateModels(holdModels);
-            models = holdModels.subList(0, 5);
+            modelsList.add(jcrContent.adaptTo(NewsCardModel.class));
         }
+        return modelsList;
     }
 
     public String getPath() {
         return resource.getPath();
     }
 
-    private List<NewsCardModel> separateModels(List<NewsCardModel> list, int paginationNumber) {
-        int limit = Math.min(paginationNumber * 5, list.size());
-        List<NewsCardModel> separatedPart = new ArrayList<>();
-        for (int i = (paginationNumber - 1) * 5; i < limit; i++) {
-            separatedPart.add(list.get(i));
-        }
-        return separatedPart;
-    }
-
-    private void paginateModels(List<NewsCardModel> models) {
-        double listSize = models.size();
-        double listPart = listSize / 5;
-        double roundPart = Math.ceil(listPart);
-
-        pageNumbers = IntStream.range(1, (int) (roundPart + 1)).boxed().collect(Collectors.toList());
-
-    }
-
-    private void isNumeric(String string) {
-        int intValue;
-
-        if (string == null || string.equals("")) {
-            log.info("String cannot be parsed, it is null or empty.");
-        }
-
+    private List<NewsCardModel> separateModels(List<NewsCardModel> list, String paginationNumber) {
+        int pNumber = 1;
         try {
-            intValue = Integer.parseInt(string);
+            pNumber = Integer.parseInt(paginationNumber);
         } catch (NumberFormatException e) {
-            log.error("Input String cannot be parsed to Integer.");
+            log.error("Exception occurred while parsing request parameter " + e.getMessage());
         }
+        pNumber = (pNumber - 1) * CARDS_PERPAGE;
```

Comment:
(code style) It is bad to use all the same variable for different purposes.
At the beginning of this method. `pNumer` is the number of the page. Like 1, 2, 3.
Here `pNumber` is the cout of cards before the current page. This must be a different variable.